lib/encoding/zstd/stream_pure.go

```diff
@@ -0,0 +1,81 @@
+//go:build !cgo
+
+package zstd
+
+import (
+	"io"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"github.com/klauspost/compress/zstd"
+)
+
+// Reader is zstd reader
+type Reader struct {
+	d *zstd.Decoder
+}
+
+// NewReader returns zstd reader for the given r.
+func NewReader(r io.Reader) *Reader {
+	d, err := zstd.NewReader(r)
+	if err != nil {
+		logger.Panicf("BUG: failed to create ZSTD reader: %s", err)
+	}
+	return &Reader{
+		d: d,
+	}
+}
+
+// Close implements io.ReadCloser interface
+func (r *Reader) Close() error {
+	r.Release()
+	return nil
+}
+
+// Read reads up to len(p) bytes to p from r.
+func (r *Reader) Read(p []byte) (int, error) {
+	return r.d.Read(p)
+}
+
+// Reset updates supplied stream r.
+func (r *Reader) Reset(reader io.Reader) {
+	_ = r.d.Reset(reader)
+}
+
+// Release releases r.
+func (r *Reader) Release() {
+	r.d.Close()
```

Comment:
The same as goztd:
https://pkg.go.dev/github.com/klauspost/compress/zstd#Decoder.Reset
https://pkg.go.dev/github.com/klauspost/compress/zstd#Decoder.Close